src/pages/api/gamer/[platform]/[username].ts

```diff
@@ -0,0 +1,122 @@
+                username: username,
+                platform: platform
+            }
+        },
+        'teammate_analysis': {
+            query: {
+                shooting_player: req.query.username,
+                shooting_player_platform: req.query.platform
+            }
+        },
+        'gamer_stats_graded': {
+            query: {
+                username: req.query.username,
+                platform: req.query.platform
+            }
+        },
+        'gamers': {
+            query: {
+                username: req.query.username,
+                platform: req.query.platform
+            }
+        },
+        'time_of_day_analysis': {
+            query: {
+                timezone: 'America/Los_Angeles',
+                username: req.query.username,
+                platform: req.query.platform,
+                cutoff: "10"
+            }
+        },
+        'day_of_week_analysis': {
+            query: {
+                timezone: 'America/Los_Angeles',
+                username: req.query.username,
+                platform: req.query.platform,
+                cutoff: "10"
+            }
+        }
+    };
+    let gamerQueryData = await queryGamers({username: username, platform: platform});
+    let gamerData =gamerQueryData[0]
+    if(gamerData){
+        let promises = Object.keys(views).filter((v)=> defaultViews.includes(v) || queriedViews.includes(v)).map(async (key: string) => await queryView(key, views[key].query));
+        Bluebird.all(promises).then(async (arrData) => {
+            Object.keys(views).filter((v)=> defaultViews.includes(v) || queriedViews.includes(v)).forEach((key: string, index: number) => views[key].data = arrData[index]);
+            let gamer = sanitizeGamer(views['player_stat_summary']['data'][0]) || {};
+            let gamerPromise = Bluebird.resolve(gamerData);
+            let viewData = views[view as string]['data'];
+            if(view as string === 'teammate_analysis'){
+                let newViewData = {};
+                newViewData['teammates']= sanitizeTeammates(viewData);
+                newViewData['titleKeys'] =  Object.keys(viewData[0]).filter((key) => !FILTER_KEYS.includes(key)).map(columnToDisplayName);
+                viewData = newViewData;
+            }
+            if (!gamerData.needs_update) {
+                gamerData.needs_update = true;
+                gamerPromise = updateGamer({username: gamerData.username, platform: gamerData.platform}, gamerData);
+            }
+            await gamerPromise;
+            let seoMetadata = {
+                title: 'Warzone stats for ' + gamer.username,
+                keywords: ['warzone', 'stats', 'kdr', 'gulag wins'],
+                description: 'KDR: ' + gamer.kdr + ' Gulag Win Rate: ' + gamer.gulag_win_rate
+            };
+
+            res.json({
+                gamer: gamer,
+                viewData: viewData,
+                seoMetadata: seoMetadata,
+                filterKeys: FILTER_KEYS
```

Comment:
What are the filterKeys?

Reply:
Those actually aren't needed since I do the filtering upstream.

Reply:
What do you think about something to standardize the view data structure?

Instead of {gamer, viewData, seoMetadata, filterKeys}

we did something like {data: {gamer, teammates}, error: {message, type}, metadata: {}, options: {}, params: {}}

or something of the sort? make it standard across all the views?